File: backend/app/interfaces/schemas/response.py
from typing import Any, Generic, Optional, TypeVar, List
from datetime import datetime
from pydantic import BaseModel
from app.interfaces.schemas.event import AgentSSEEvent
from app.domain.models.session import SessionStatus

T = TypeVar('T')


class APIResponse(BaseModel, Generic[T]):
    code: int = 0
    msg: str = "success"
    data: Optional[T] = None

    @staticmethod
    def success(data: Optional[T] = None) -> "APIResponse[T]":
        return APIResponse(code=0, msg="success", data=data)

    @staticmethod
    def error(code: int, msg: str) -> "APIResponse[T]":
        return APIResponse(code=code, msg=msg, data=None)


class CreateSessionResponse(BaseModel):
    session_id: str


class GetSessionResponse(BaseModel):
    session_id: str
    title: Optional[str] = None
    events: List[AgentSSEEvent] = []


class ListSessionItem(BaseModel):
    session_id: str
    title: Optional[str] = None
    latest_message: Optional[str] = None
    latest_message_at: Optional[int] = None
    status: SessionStatus
    unread_message_count: int


class ListSessionResponse(BaseModel):
    sessions: List[ListSessionItem]


class ConsoleRecord(BaseModel):
    ps1: str
    command: str
    output: str


class ShellViewResponse(BaseModel):
    output: str
    session_id: str
    console: Optional[List[ConsoleRecord]] = None


class FileViewResponse(BaseModel):
    content: str
    file: str


class AttachmentUploadResponse(BaseModel):
    filename: str
    content_type: str
    file_size: int
    storage_type: str
    storage_url: str


class AttachmentDownloadResponse(BaseModel):
    storage_url: str
    filename: str
    content_type: str
    content: str
    file_size: int


class SessionAttachmentsResponse(BaseModel):
    session_id: str
    attachments: List[dict]

File: frontend/src/types/response.ts
import { AgentSSEEvent } from "./event";

export enum SessionStatus {
    PENDING = "pending",
    RUNNING = "running",
    WAITING = "waiting",
    COMPLETED = "completed"
}

export interface CreateSessionResponse {
    session_id: string;
}

export interface GetSessionResponse {
    session_id: string;
    title: string | null;
    events: AgentSSEEvent[];
}

export interface ListSessionItem {
    session_id: string;
    title: string | null;
    latest_message: string | null;
    latest_message_at: number | null;
    status: SessionStatus;
    unread_message_count: number;
}

export interface ListSessionResponse {
    sessions: ListSessionItem[];
}

export interface ConsoleRecord {
    ps1: string;
    command: string;
    output: string;
  }
  
  export interface ShellViewResponse {
    output: string;
    session_id: string;
    console: ConsoleRecord[];
  }

export interface FileViewResponse {
    content: string;
    file: string;
  }
  